sqlglot/generator.py

```diff
@@ -1394,6 +1394,11 @@ def bytestring_sql(self, expression: exp.ByteString) -> str:
                 delimiter=self.dialect.BYTE_END,
                 escaped_delimiter=self._escaped_byte_quote_end,
             )
+            is_bytes = expression.args.get("is_bytes")
+            if is_bytes and is_bytes != self.dialect.BYTE_STRING_IS_BYTES_TYPE:
+                return self.sql(
```

Comment:
What about the reverse direction, i.e., DuckDB to BigQuery? Should be a small lift to handle:

```python
is_bytes = expression.args.get("is_bytes", False)
if is_bytes and not self.dialect.BYTE_STRING_IS_BYTES_TYPE:
    # cast to varchar and return result
if not is_bytes and self.dialect.BYTE_STRING_IS_BYTES_TYPE:
    # cast to binary and return result
...
```